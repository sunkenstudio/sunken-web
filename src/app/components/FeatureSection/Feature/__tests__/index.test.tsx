'use client';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Feature } from '..';

describe('Feature', () => {
  it('match snapshot', () => {
    const rendered = render(
      <Feature
        type={'headset'}
        subHeader={'this is the subheader'}
        blurb={'this is the blurb'}
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  it('if given subheader, subheader should render', () => {
    const subheader = 'this is the subheader';
    const rendered = render(
      <Feature
        type={'headset'}
        subHeader={subheader}
        blurb={'this is the blurb'}
      />
    );
    expect(rendered.getByText(subheader)).toBeDefined();
  });

  it('if given blurb, blurb should render', () => {
    const blurb = 'this is the blurb';
    const rendered = render(
      <Feature
        type="headset"
        subHeader={'this is the subheader'}
        blurb={blurb}
      />
    );
    expect(rendered.getByText(blurb)).toBeDefined();
  });

  it('if given icon type, icon should render', () => {
    const rendered = render(
      <Feature
        type="headset"
        subHeader={'this is the subheader'}
        blurb={'this is the blurb'}
      />
    );
    expect(rendered.getByTitle('headset icon')).toBeDefined();
  });

  it('if NOT given icon type, icon should NOT render', () => {
    const rendered = render(
      <Feature
        subHeader={'this is the subheader'}
        blurb={'this is the blurb'}
      />
    );
    expect(rendered.queryByTitle('headset icon')).toBeNull();
  });
});
