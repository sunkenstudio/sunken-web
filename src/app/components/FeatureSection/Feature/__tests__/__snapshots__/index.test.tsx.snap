// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1bzvqlf"
      >
        <svg
          fill="currentColor"
          height="64"
          viewBox="0 0 256 256"
          width="64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            headset icon
          </title>
          <path
            d="M201.89,54.66A103.43,103.43,0,0,0,128.79,24H128A104,104,0,0,0,24,128v56a24,24,0,0,0,24,24H64a24,24,0,0,0,24-24V144a24,24,0,0,0-24-24H40.36A88.12,88.12,0,0,1,190.54,65.93,87.39,87.39,0,0,1,215.65,120H192a24,24,0,0,0-24,24v40a24,24,0,0,0,24,24h24a24,24,0,0,1-24,24H136a8,8,0,0,0,0,16h56a40,40,0,0,0,40-40V128A103.41,103.41,0,0,0,201.89,54.66ZM64,136a8,8,0,0,1,8,8v40a8,8,0,0,1-8,8H48a8,8,0,0,1-8-8V136Zm128,56a8,8,0,0,1-8-8V144a8,8,0,0,1,8-8h24v56Z"
          />
        </svg>
        <h4
          class="chakra-heading css-amjuq9"
        >
          this is the subheader
        </h4>
        <p
          class="chakra-text css-0"
        >
          this is the blurb
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1bzvqlf"
    >
      <svg
        fill="currentColor"
        height="64"
        viewBox="0 0 256 256"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          headset icon
        </title>
        <path
          d="M201.89,54.66A103.43,103.43,0,0,0,128.79,24H128A104,104,0,0,0,24,128v56a24,24,0,0,0,24,24H64a24,24,0,0,0,24-24V144a24,24,0,0,0-24-24H40.36A88.12,88.12,0,0,1,190.54,65.93,87.39,87.39,0,0,1,215.65,120H192a24,24,0,0,0-24,24v40a24,24,0,0,0,24,24h24a24,24,0,0,1-24,24H136a8,8,0,0,0,0,16h56a40,40,0,0,0,40-40V128A103.41,103.41,0,0,0,201.89,54.66ZM64,136a8,8,0,0,1,8,8v40a8,8,0,0,1-8,8H48a8,8,0,0,1-8-8V136Zm128,56a8,8,0,0,1-8-8V144a8,8,0,0,1,8-8h24v56Z"
        />
      </svg>
      <h4
        class="chakra-heading css-amjuq9"
      >
        this is the subheader
      </h4>
      <p
        class="chakra-text css-0"
      >
        this is the blurb
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
